napari/_vispy/vispy_surface_layer.py

```diff
@@ -81,6 +85,17 @@ def _on_gamma_change(self, event=None):
     def _on_shading_change(self, event=None):
         if self.layer.shading == 'none':
             self.node.shading = None
+            if self.node.shading_filter is not None:
+                self.node.shading_filter._attached = False
```

Comment:
@alisterburt I wonder if we should contribute these two lines upstream into vispy so they get called when `self.node.shading = None`. Would be nice to have have to go into their private stuff

Reply:
great idea, will do!